File: test_client.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import time
import sys

from net import ImageMultiClient

def main():
    servers = [(("localhost", 2204), 10.0),
               (("localhost", 2205), 10.0)]
    client = ImageMultiClient(servers)
    while True:
        client.write_status()
        time.sleep(0.3)
        while True:
            #print "advance"
            res = client.advance_to_stream(0)
            if not res:
                break

if __name__ == '__main__':
    main()

File: test_client.cpp

#include "net.hpp"

using namespace chrono;

int main() {

  vector< ImageServerConf > servers;
  servers.emplace_back("localhost", "2204", 2.0);
  servers.emplace_back("localhost", "2205", 2.0);
  ImageMultiClient client(servers);

  while (true) {
    while (true) {
      auto res = client.advance_to_stream(0, false);
      if (res.size() == 0) break;
      else {
        for (auto &r : res) {
          fprintf(stderr, "%f  ", r->timestamp);
        }
        fprintf(stderr, "\n");
      }
    }
    client.print_status();
    this_thread::sleep_for(milliseconds(300));
  }

  return 0;

}
